server/routerlicious/packages/services-ordering-zookeeper/src/zookeeperClient.ts

```diff
@@ -38,13 +31,16 @@ export class ZookeeperClient implements IZookeeperClient {
     }
 
     private connect() {
-        this.client = zookeeper.createClient(this.url, { retries: RetryAttemps });
-        this.client.connect();
-        this.client.once("connected", () => {
-            this.client.once("disconnected", () => {
+        this.client = new ZooKeeper({
+            connect: this.url,
+            timeout: 30000,
+        });
+        this.client.once("connect", () => {
+            this.client.once("close", () => {
```

Comment:
It looks like if we call close it will just reconnect. We may want to add a `closed` flag and set it to true when `ZookeeperClient.close` is called?

Reply:
When we call close(), we are removing all listeners, which will destroy this listener here. Had the same thought though before realizing that :)